### DISCUSSION TITLE
Alpine Headless Listbox within a Livewire environment

### DISCUSSION THREAD

**Discussion by robdekort** [livewire/livewire#8942] [1. Help]
Hello there. I'm trying to use the Headless Listbox component as per this example: https://alpinejs.dev/component/headless-listbox

However, I'm working in a Livewire environment so my bundle looks like this:

```js
import { Livewire, Alpine } from '../../vendor/livewire/livewire/dist/livewire.esm'
import precognition from 'laravel-precognition-alpine'
import ui from '@alpinejs/ui'

Alpine.plugin(precognition, ui)
Livewire.start()
```

Livewire and basic components work just fine. However with the listbox I get the following errors: `ReferenceError: Can't find variable: $listboxOption`

<img width="742" alt="Screenshot 2024-10-16 at 15 51 16" src="https://github.com/user-attachments/assets/20e05902-ee26-4cc3-8ed4-f261ebb5ab24">

I'm wondering how to load in the Alpine Components properly to prevent this from happening. If anyone has any thoughts, much appreciated. 

_Note:_ I posted this on the Alpine discussion board, but they suggest this is a livewire issue instead.

**Comment by EdgarsJoja** [comment]
It is unclear whether you are using PHP or JS variable here....

PHP variables start with `$`, so it could be reasonable to assume that it is PHP variable, in which case it's usage is not correct. `x-show` attribute requires JS expression -> this is not a JS expression, however. The correct usage (one of) would be:
```blade
x-show="{{ Js::from($listboxOption.isSelected) }}"
```
for example.
But then again, if it is a PHP variable why use alpine there at all, since it is a render time execution. Regular blade if/else would suffice:
```blade
@if($listboxOption.isSelected)
  ...
@endif
```

Now, if it actually is a JS variable, then afaik it should work. Provided, that it is within the current alpine scope. Perhaps more code would help here in this case, showing the actual area of usage.

**Reply by robdekort** [reply]
Hey sorry I wasn't more clear. I understand the confusion. It's a JS variable. I've copied the example listbox from the paid headless alpine components. The component works when I load in Alpine regularly, as per the example. But it fails when I use it together with Livewire. I'm suspecting this is a bug in Livewire.
